__tests__/app.test.js

```diff
@@ -111,4 +112,192 @@ describe('GET requests', () => {
+                expect(body.msg).toBe('Please provide username.')
+            })
+        })
+
+    })
+});
+
+describe('PATCH', () => {
+
+    describe.only('PATCH /api/articles/:article_id', () => {
+        test('201 - responds with the article votes updated correctly', () => {
+            const patchInput = { inc_votes : 20 };
+
+            return request(app)
+                .patch('/api/articles/6')
+                .send(patchInput)
+                .expect(201)
+                .then(({body}) => {
+                    const articleObj = {
+                            article_id: 6,
+                            title: 'A',
+                            topic: 'mitch',
+                            author: 'icellusedkars',
+                            body: 'Delicious tin of cat food',
+                            created_at: '2020-10-18T01:00:00.000Z',
+                            votes: 20,
+                            article_img_url: 'https://images.pexels.com/photos/158651/news-newsletter-newspaper-information-158651.jpeg?w=700&h=700'
+                    }
+                    expect(body.article).toMatchObject(articleObj)
+                })
+        });
+        test('201 - decrements votes for the article correctly', () => {
+            const patchInput = { inc_votes : -20 };
+
+            return request(app)
+                .patch('/api/articles/6')
+                .send(patchInput)
+                .expect(201)
+                .then(({body}) => {
+                    const articleObj = {
+                            article_id: 6,
+                            title: 'A',
+                            topic: 'mitch',
+                            author: 'icellusedkars',
+                            body: 'Delicious tin of cat food',
+                            created_at: '2020-10-18T01:00:00.000Z',
+                            votes: -20,
+                            article_img_url: 'https://images.pexels.com/photos/158651/news-newsletter-newspaper-information-158651.jpeg?w=700&h=700'
+                    }
+                    expect(body.article).toMatchObject(articleObj)
+                })
+        });
+        test('404 - article_id not found', () => {
+            const patchInput2 = { inc_votes : 20 };
+            return request(app)
+                .patch('/api/articles/999')
+                .send(patchInput2)
+                .expect(404)
+                .then(({body}) => {
+                    expect(body.msg).toBe('No articles found for article_id: 999')
+                })
+        });
+        test('400 - invalid input', () => {
+            const patchInput2 = { inc_votes : "cat" };
+            return request(app)
+                .patch('/api/articles/6')
+                .send(patchInput2)
+                .expect(400)
+                .then(({body}) => {
+                    expect(body.msg).toBe('invalid request')
+                })
+        })
+        test('400 - no request body.', () => {
```

Comment:
With no request body, I would return a 200 with the original article returned unchanged. This will show the request was fine (200) but nothing changed. Remember our first happy path is also a 200.

Reply:
I did this with checking the empty object 

    if(Object.keys(update).length === 0) {
        update = { inc_votes : 0 };
    }